### PR TITLE
Add transition test case of non-empty pre-state `historical_roots`

### PR DESCRIPTION
**Pull request by hwwhww**
### Issue

To cover https://github.com/prysmaticlabs/prysm/pull/13477 with this test case.

### Description
- Add fork transition test case of non-empty pre-state `historical_roots`
- [non-substantive] Update transition test format
- [non-substantive] Fix Capella spec comment typo

### CODE REVIEW

File: tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py [hunk 1]
```diff
@@ -418,3 +418,37 @@ def test_transition_with_no_attestations_until_after_fork(state, fork_epoch, spe
...
+@with_fork_metas([ForkMeta(pre_fork_name=pre, post_fork_name=post, fork_epoch=2) for pre, post in ALL_PRE_POST_FORKS])
+def test_non_empty_historical_roots(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Test with non-empty pre-state `state.historical_roots`.
+
+    Since Capella froze `historical_roots`, Capealla spec doesn't invoke `process_historical_roots_update` anymore.
+    Therefore, we need to fill in `historical_roots` with non-empty value.
+    """
+    # fill in historical_roots with non-empty values
+    pre_historical_roots = [b'\x56' * 32 for _ in range(spec.SLOTS_PER_HISTORICAL_ROOT)]
```
Comment (djrtwo) [inline tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py:432]: `SLOTS_PER_HISTORICAL_ROOT` is fine here but it's a bit misleading in it's relationship to `historical_roots` which  is a *list* of unbound length

**Review by djrtwo — APPROVED** [review]
Looks good. a quick comment on the usage of a var for the list length

File: tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py [hunk 2]
```diff
@@ -418,3 +418,37 @@ def test_transition_with_no_attestations_until_after_fork(state, fork_epoch, spe
...
     yield "blocks", blocks
     yield "post", state
+
+
+@with_fork_metas([ForkMeta(pre_fork_name=pre, post_fork_name=post, fork_epoch=2) for pre, post in ALL_PRE_POST_FORKS])
+def test_non_empty_historical_roots(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Test with non-empty pre-state `state.historical_roots`.
+
+    Since Capella froze `historical_roots`, Capealla spec doesn't invoke `process_historical_roots_update` anymore.
```
Comment (dapplion) [inline tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py:428]:
```suggestion
    Since Capella froze `historical_roots`, Capella spec doesn't invoke `process_historical_roots_update` anymore.
```

File: tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py (hunk 1 above)
Comment (dapplion) [reply tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py:432]: The test would do its job with > 0 items so why no set it to 1?

Comment (hwwhww) [inline tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py:432]:
```suggestion
    pre_historical_roots = [b'\x56' * 32]
```

**Comment by hwwhww** [comment]
@djrtwo @dapplion thank you for you reviews! I addressed your feedback.
